pretyping/nativenorm.ml

```diff
@@ -421,7 +421,7 @@ and nf_evar env sigma evk args =
     (* nf_args takes arguments in the reverse order but produces them
        in the correct one, so we have to reverse them again for the
        evar node *)
-    mkEvar (evk, List.rev args), ty
+    mkEvar (evk, SList.of_full_list @@ List.rev args), ty
```

Comment:
```suggestion
    mkLEvar (evk, List.rev args), ty
```

Reply:
Doens't type-check, this is in constr land.

Reply:
So you need to add a few casts

Reply:
It's actually annoying to write, which is why I didn't do it in the first place. It doesn't seem to happen often (read: virtually never) from my runtime experiments, I'll leave that to another PR.

Reply:
```suggestion
    EConstr.(Unsafe.to_constr @@ mkLEvar sigma (evk, List.rev_map of_constr args)), ty
```